File: LeetCode/Easy/1342_Number_of_Steps_to_Reduce_a_Number_to_Zero/Solution.java
// 2021-02-12
class Solution {
    public int numberOfSteps (int num) {
        int answer = 0;
        
        while (num > 0) {
            if ((num % 2) == 0) {
                num /= 2;
                answer++;
            } else {
                --num;
                answer++;
            }
        }
        return answer;
    }
}

File: LeetCode/Easy/1342_Number_of_Steps_to_Reduce_a_Number_to_Zero/Solution.cpp
// 2022-05-26
class Solution {
public:
    int numberOfSteps(int num) {
        int count = 0;
        while (num > 0) {
            if (num%2 == 0) 
                num = num >> 1;
            else
                num -= 1;
            count++;
        }
        return count;
    }
};